src/lib/QtWidgets/QTabWidget/index.ts

```diff
@@ -0,0 +1,56 @@
+import addon from "../../core/addon";
+import { NodeWidget } from "../QWidget";
+import { BaseWidgetEvents } from "../../core/EventWidget";
+import { NativeElement } from "../../core/Component";
+import { QIcon } from "../../QtGui/QIcon";
+import { TabPosition } from "../../QtEnums";
+
+export const QTabWidgetEvents = Object.freeze({
+  ...BaseWidgetEvents,
+  currentChanged: "currentChanged",
+  tabBarClicked: "tabBarClicked",
+  tabBarDoubleClicked: "tabBarDoubleClicked",
+  tabCloseRequested: "tabCloseRequested"
+});
+
+export class QTabWidget extends NodeWidget {
+  native: NativeElement;
+  constructor(parent?: NodeWidget) {
+    let native;
+    if (parent) {
+      native = new addon.QTabWidget(parent.native);
+    } else {
+      native = new addon.QTabWidget();
+    }
+    super(native);
+    this.parent = parent;
+    this.native = native;
+    // bind member functions
+    this.addTab.bind(this);
```

Comment:
I think its better to bind remaining methods here as well 😬

Reply:
@master-atul Was quite bussy last few days, will tackle this issue this weekend (along with the missing widget documentation 😊 )

Reply:
Take it easy 😄

Reply:
Hey @Uriziel01  I am going to merge this anyway, You can maybe create a separate PR for docs. I need this to go in before I can merge my PR: native plugins one (which contains a lot of core changes)

Reply:
Or I think I ll just leave it be for few more days 😬

Reply:
Why even bind member functions? The context will always be the instance of the class unless called with a different context.
Same goes for methods as arrow functions, it's completely unnecessary and really fucks up class declarations.

Reply:
Hey @marcus-sa 
I thinking binding is helpful in some edge cases like this where the context is different like you said:
```
class MyWidget {
  constructor(name) {
    this.native = { name };
    this.myWidgetFunction = this.myWidgetFunction.bind(this);
  }
  myWidgetFunction() {
    console.log(this.native.name, " is my name");
  }
}

// User code
const instance = new MyWidget("fancy widget");
const { myWidgetFunction } = instance;
// somewhere else in the app
myWidgetFunction();
```

I dont think we are binding the functions in case of arrow functions anywhere. Did you find any ?

Also, just curious on what do you really mean by `fucks up class declarations`? Is it the one extra line or something else that I missed?

Reply:
If you try to extend classes from `@nodegui/nodegui` and override any arrow member functions using methods, it'll throw an error as I've already described in a previous [issue](https://github.com/nodegui/nodegui/issues/44)

TL;DR
![test](https://camo.githubusercontent.com/f25c848871b294e7e69b7e584fb42d965d07c293/68747470733a2f2f692e696d6775722e636f6d2f346c63314c58452e706e67)

Reply:
Yes but this project is not only for Typescript, It can be used in regular javascript aswell. 

But do you think it would cause any other issue than just a single extra line in the code base for each function ?

Reply:
Oh alright I see what you mean @marcus-sa

Reply:
Had to read it twice. 
Okay so in the case of overloading, in typescript you would need to just declare the function upfront that was in the base class.

For example
```
export class RNScrollArea extends QScrollArea  {
  setWidget(widget: NodeWidget): void; //this is from QScrollArea
  setWidget() { 
    console.log("hello");
  }
}
```

Reply:
No, the problem is that you define class methods as member properties (arrow functions) instead of member functions (non arrow functions), which will cause the compiler to throw an error if I don't override it using arrow functions aswell.

Reply:
I dont agree to your points here. Arrow functions are useful in automatic binding the context whenever you use this inside the member functions. And as i mentioned above it allows in handling certain edge cases. 

I dont see any issue with the code inconsistencies you are mentioning. It is fairly straightforward and easier to manage at this point. You are free to fork this project and change it according to your needs as i had mentioned before.